### PR TITLE
feat(operator): add Model selector for scale subresource to enable HPA-based scaling

### PR DESCRIPTION
**Pull request by lc525**
- updates the Model CRD to contain a pod selector in the scale subresource
- sets the selector to a label `server=[inference-server-name]` matching no actual pods
- docs [to be moved to gitbook before merging]

**Which issue(s) this PR fixes**:
Fixes #1190 (internal): allow HPA-based Model autoscaling

**Special notes for your reviewer**:
- docs added to the PR to show how HPA would work
- tested in kind (small-scale)

**TODO**:
- [ ] test availability during HPA scale-up/scale-down via k6 load test (can be done after merging)

### CODE REVIEW

File: k8s/helm-charts/seldon-core-v2-crds/templates/seldon-v2-crds.yaml
Comment (lc525) [inline k8s/helm-charts/seldon-core-v2-crds/templates/seldon-v2-crds.yaml:1]: generated

File: k8s/yaml/crds.yaml
Comment (lc525) [inline k8s/yaml/crds.yaml:1]: generated

File: k8s/yaml/runtime.yaml
Comment (lc525) [inline k8s/yaml/runtime.yaml:1]: generated

File: k8s/yaml/servers.yaml
Comment (lc525) [inline k8s/yaml/servers.yaml:1]: generated

File: operator/config/crd/bases/mlops.seldon.io_models.yaml
Comment (lc525) [inline operator/config/crd/bases/mlops.seldon.io_models.yaml:1]: generated

**Review by sakoush — APPROVED** [review]
LGTM

**Pull request merged**